### ISSUE TITLE
Saving object with CLP Update-only throws error

### ISSUE THREAD

**Issue by rihadavid** [parse-community/parse-server#5177]
### Issue Description

When saving object from client side with CLP set to Update-only, it throws `Permission denied for action get on class Comment.` When I allow Get in CLP, it starts working again. I tried to log inside `Parse.Cloud.beforeSave("Comment")`, but it does not even get called.

It is happening on 2.8.4, tried also 2.6.5, 2.8.0 and 2.8.3, those are ok. Did not try newer, but did not see this issue mentioned in the changelog.

### Steps to reproduce

Set CLP to Update allowed, everything else disabled. Create an object in the dashboard and then try to update any of its fields from the client side, without the master key.
 
### Expected Results

Object should be saved, because we have Update allowed in CLP.

### Actual Outcome

Object is not saved, although we have Update allowed in CLP. We also need Get allowed.

### Environment Setup

- **Server**
  - parse-server version: 2.8.4
  - Operating System: WINDOWS 10
  - Localhost or remote server: LOCAL

- **Database**
  - MongoDB version: 3.6.6
  - Localhost or remote server? mLab

### Logs/Trace

```
verbose: REQUEST for [PUT] /parse/classes/Comment/bIRLdNysQH: {
  "temp": "!likes"
} method=PUT, url=/parse/classes/Comment/bIRLdNysQH, x-parse-application-id=parselocal, x-parse-client-version=net-xamarin-android1.7.0.0, x-parse-installation-id=26e8ef64-ccfa-420b-a475-fe8663bf555c, x-parse-app-build-version=25, x-parse-app-display-version=1.7.7, x-parse-os-version=6.0.1, x-parse-windows-key=, x-parse-session-token=r:f8591e34b5922feb142a388fb580c33c, content-type=application/json, if-modified-since=Fri, 16 Nov 2018 12:29:54 GMT, content-length=17, expect=100-continue, host=10.0.0.3:1337, temp=!likes
error: Error generating response. ParseError {
  code: 119,
  message: 'Permission denied for action get on class Comment.' } code=119, message=Permission denied for action get on class Comment.
error: Permission denied for action get on class Comment. code=119, message=Permission denied for action get on class Comment.
```

**Comment by rihadavid** [comment]
Dear stalebot, please do not close this issue, it seems to be an easily reproducible bug ⚔️ 😄 
Would anyone like to try to reproduce this on the newest version? Maybe @flovilmart or @dplewis ? I have unfortunately migrated my app to a provider which is using parse server 2.3.3, so it does not bother me right now, but you might want to take a look before the stalebot decides it's not worth it 😉

**Comment by flovilmart** [comment]
Did you try the latest version of parse server? If I recall correctly a fix was addressed for this

**Comment by rihadavid** [comment]
No, the newest I tried was 2.8.4

**Comment by flovilmart** [comment]
Can you try with the latest version?

**Comment by rihadavid** [comment]
Sorry, I can't test it right now, I am on 2.3.3 as I said

**Comment by rihadavid** [comment]
I have just discovered, that this bug is surfacing on 2.8.4 - 3.1.3 also in another case. It's already fixed, but if any of you are using 2.8.4 - 3.1.3, which is currently offered for example on Sashido, you might want to know about this.

Let's say you have a class with `Public CLP Update-only` + `_User-Pointer CLP Read-enabled`.
You'd expect that any user can update objects in this class, but if you try to update the object as a user who is not the one in the `_User-Pointer CLP` field, the object is updated successfully but in `beforeSave` hook, the request.object is an empty pointer and has no data. This might very well break your logic.

If you want to see some code when this is happening:
```
Parse.Cloud.define("test", async (request) => {
    //create a Test class with fields string:txt, bool:allowed, _User-Pointer:author
    //before calling this, set 'allowed' field of the object to true in the dashboard
    //don't forget to use `Public CLP Update-only` + `_User-Pointer CLP Read-enabled`
    var senderUser = request.user; //call the function as any other user than in your '_User-Pointer' field
    var Test = Parse.Object.extend("Test");
    let test = new Test();
    test.id = "ppQEBcxDrE"; //use the id of your object
    test.set("txt", "updated");
    await Parse.Object.saveAll([test], { sessionToken: senderUser.getSessionToken() });
});

Parse.Cloud.beforeSave("Test", async (request) => {
    var testToSave = request.object;
    if (!testToSave.get("allowed"))
        throw "allowed is false"; //this will be thrown, although your object has 'allowed' set to true
});
```
